dat/missions/trader/anxiousmerchant.lua

```diff
@@ -111,8 +111,9 @@ function land()
          player.pay(mem.payment)
       else
-         tk.msg(_("Deliver the Goods... late"), _([[Landing at the spaceport you see the Traders Guild depot surrounded by a fraught hum of activity. The cargo inspector looks at you with surprise and then anger, "What the hell is this?! This shipment was supposed to be here ages ago! We've been shifting stuff around to make up for it and then you come waltzing in here... where the hell is the employee who was supposed to deliver this stuff?" A group of workers rushes along with you and the inspector and you as you try to explain what happened. "That fool has been causing us all sorts of problems, and passing on the job to someone as incompetent as you is the last straw! I swear!"
-    You wait to one side as the cargo is hauled off your ship at breakneck speed and wonder if you should have just dumped the stuff in space. Just as the last of the cargo is taken off your ship, the inspector, who has clearly cooled off a bit, comes up to you and says "Look, I know you were trying to do us a favour but next time don't bother if you can't make it on time. I'm glad you didn't just dump it all into space like some people have done, but I can't pay you for this." He shakes his head and walks away. "That pilot is so fired...."]]))
+         faction.modPlayerSingle("Traders Guild", -1)
+         tk.msg(_("Deliver the Goods… late"), _([[Landing at the spaceport you see the Traders Guild depot surrounded by a fraught hum of activity. The cargo inspector looks at you with surprise and then anger, "What the hell is this?! This shipment was supposed to be here ages ago! We've been shifting stuff around to make up for it and then you come waltzing in here… where the hell is the employee who was supposed to deliver this stuff?" A group of workers rushes along with you and the inspector and you as you try to explain what happened. "That fool has been causing us all sorts of problems, and passing on the job to someone as incompetent as you is the last straw! I swear!"
+    You wait to one side as the cargo is hauled off your ship at breakneck speed and wonder if you should have just dumped the stuff in space. Just as the last of the cargo is taken off your ship, the inspector, who has clearly cooled off a bit, comes up to you and says "Look, I know you were trying to do us a favour but next time don't bother if you can't make it on time. I'm glad you didn't just dump it all into space like some people have done, but I can't pay you for this." He shakes his head and walks away. "That pilot is so fired…"]]))
       end
       misn.cargoRm(mem.cargo_ID)
```

Comment:
No need to remove the cargo as all mission cargos get removed when the associated mission "finishes" with `misn.finish()`.

Reply:
Yeah, that's a left over... will be fixed in a moment :)

Can I ask if we have a mission template or is there one I can use as such so I can learn what the basic format of a mission should be, at least it would mean I'm a little more sure of of things like this and the "misn.finish(false) -> return" thing, though I have to admit I'm not entirely clear on that.... my current working hypothesis is that return should be used when you want to allow the player to talk to the mission giver again but misn.finish(false) removes this instance of the mission (which for the cargo missions I felt was just the way it worked rather than necessarily an issue - though it was a little annoying for cargo missions with a time limit since a ship switch might be possible but then the missions disappeared. So I guess I'm on the fence in that respect! :)

Reply:
There is a template at https://github.com/naev/naev/blob/main/docs/missions/mission_template.lua . However, looking at the code of any recent mission should also work.

> my current working hypothesis is that return should be used when you want to allow the player to talk to the mission giver again but misn.finish(false) removes this instance of the mission 

Yes, that's the case. However, in general we want missions to have the following properties:
1. Allow the player to abort or fail without consequence (mission can be retaken again)
2. Don't make the mission go away if the player doesn't accept or have enough cargo

Reply:
Understood :)